lib/read/lib/Cardano/Wallet/Read/Block/Gen.hs

```diff
@@ -0,0 +1,101 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Cardano.Wallet.Read.Block.Gen
+where
+
+import Prelude hiding
+    ( (.)
+    )
+
+import Cardano.Wallet.Read
+    ( Tx (..)
+    )
+import Cardano.Wallet.Read.Block
+    ( Block (..)
+    , ConsensusBlock
+    , toConsensusBlock
+    )
+import Cardano.Wallet.Read.Block.Gen.Babbage
+    ( mkBabbageBlock
+    )
+import Cardano.Wallet.Read.Block.Gen.BlockParameters
+    ( BlockParameters (..)
+    , exampleBlockParameters
+    )
+import Cardano.Wallet.Read.Block.Gen.Byron
+    ( mkByronBlock
+    )
+import Cardano.Wallet.Read.Block.Gen.Shelley
+    ( mkShelleyBlock
+    )
+import Cardano.Wallet.Read.Eras
+    ( (:.:) (..)
+    , EraFun (..)
+    , extractEraValue
+    , sequenceEraValue
+    , (*.**)
+    )
+import Cardano.Wallet.Read.Eras.EraFun
+    ( AllEraValue (..)
+    , runAllEraValue
+    )
+import Cardano.Wallet.Read.Tx.Gen.Allegra
+    ( exampleAllegraTx
+    )
+import Cardano.Wallet.Read.Tx.Gen.Alonzo
+    ( exampleAlonzoTx
+    )
+import Cardano.Wallet.Read.Tx.Gen.Byron
+    ( exampleByronTx
+    )
+import Cardano.Wallet.Read.Tx.Gen.Mary
+    ( exampleMaryTx
+    )
+import Cardano.Wallet.Read.Tx.Gen.Shelley
+    ( exampleShelleyTx
+    )
+import Control.Category
+    ( (.)
+    )
+
+mkBlockEra :: EraFun BlockParameters Block
+mkBlockEra =
+    EraFun
+        { byronFun = g mkByronBlock
+        , shelleyFun = g mkShelleyBlock
+        , allegraFun = g mkShelleyBlock
+        , maryFun = g mkShelleyBlock
+        , alonzoFun = g mkShelleyBlock
+        , babbageFun = g mkBabbageBlock
+        , conwayFun = g mkBabbageBlock
```

Comment:
I would align this with mainnet protocol versions. This requires adding a parameter to `mkShelleyBlock` and `mkBabbageBlock`. The protocol versions are

```
Shelley	— 2.0
Allegra	— 3.0
Mary	— 4.0
???	— 5.0
Alonzo	— 6.0
Babbage	— 7.0
Conway	— 8.0
```